src/search/view/SearchView.ts

```diff
@@ -232,6 +243,27 @@ export class SearchView extends BaseTopLevelView implements TopLevelView<SearchV
 		this.viewSlider = new ViewSlider([this.folderColumn, this.resultListColumn, this.resultDetailsColumn])
 	}
 
+	private async routeToType(type: string): Promise<void> {
+		let latestRestriction = null
+		// contacts do not have restrictions at this time
+		if (type === "mail") {
+			latestRestriction = this.searchViewModel.latestMailRestriction
+		} else if (type === "calendar") {
+			latestRestriction = this.searchViewModel.latestCalendarRestriction
+		}
+		const searchRouter = await locator.scopedSearchRouter()
+		if (this.searchViewModel.currentQuery) {
```

Comment:
I think this logic should be in the viewModel. To have more than one `if` is already a lot for the poor, busy view. There are other signs too e.g. that you need to expose latest restriction and router. Instead I'd create a method like `switchToSearchType()` or something like that. We might still need to call viewSlider here though